tests/ws/driver_test.go

```diff
@@ -0,0 +1,453 @@
+package ws
+
+import (
+	"context"
+	"database/sql"
+	"os"
+	"testing"
+	"time"
+
+	_ "github.com/libsql/libsql-client-go/libsql"
+)
+
+func Test_QueryExec(t *testing.T) {
+	dbURL := os.Getenv("LIBSQL_TEST_DB_URL")
+	db, err := sql.Open("libsql", dbURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.Background()
+	_, err = db.ExecContext(ctx, "CREATE TABLE IF NOT EXISTS test (id INTEGER PRIMARY KEY, name TEXT)")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = db.ExecContext(ctx, "INSERT INTO test (name) VALUES (?)", "hello world")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rows, err := db.QueryContext(ctx, "SELECT * FROM test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var id int64
+		var name string
+		err = rows.Scan(&id, &name)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if id != 1 {
+			t.Fatal("id should be 1")
+		}
+		if name != "hello world" {
+			t.Fatal("name should be hello world")
+		}
+	}
+
+	err = rows.Err()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		_, err = db.ExecContext(ctx, "DROP TABLE test")
+		if err != nil {
+			t.Fatal(err)
+		}
+		db.Close()
+	})
+}
+
+func Test_Transactions(t *testing.T) {
+	dbURL := os.Getenv("LIBSQL_TEST_DB_URL")
+	db, err := sql.Open("libsql", dbURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := context.Background()
+	_, err = db.ExecContext(ctx, "CREATE TABLE IF NOT EXISTS test (id INTEGER PRIMARY KEY, name TEXT)")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = tx.ExecContext(ctx, "INSERT INTO test (name) VALUES (?)", "hello world")
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rows, err := db.QueryContext(ctx, "SELECT * FROM test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var id int64
+		var name string
+		err = rows.Scan(&id, &name)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if id != 1 {
+			t.Fatal("id should be 1")
+		}
+		if name != "hello world" {
+			t.Fatal("name should be hello world")
+		}
+	}
+
+	err = rows.Err()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		_, err = db.ExecContext(ctx, "DROP TABLE test")
+		if err != nil {
+			t.Fatal(err)
+		}
+		db.Close()
+	})
```

Comment:
True. I left them like that, thinking it will be flexible in case some tests need to be updated in the future. But I think we can remove them now. 

Also, it removed like 100 lines, so its all good ;)

> 1 file changed, 45 insertions(+), 171 deletions(-)

Reply:
Great. Thank you!